### PR TITLE
[enhancement](cloud)avoid transaction too large when dropping partition

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

Issue Number: close #xxx

When we truncate table or drop a large number of partitions at once, it is possible to return an fdb transaction_too_large error. 

According to the [documentation](https://github.com/apple/foundationdb/wiki/Transaction-size-limit), this is due to the transaction mutated data being too large, or the transaction has too much conflict ranges with other transactions.

So we need to reconsider the design here.

We can split the step of getting partition version and put it before the drop-partitions transaction.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run feut

**Comment by xy720** [comment]
run fe_ut

File: cloud/src/meta-service/meta_service_partition.cpp [hunk 1]
```diff
@@ -600,47 +600,12 @@ void MetaServiceImpl::drop_partition(::google::protobuf::RpcController* controll
...
-            }
-            if (part_version > 1) {
-                need_update_table_version = true;
-                break;
-            }
-        }
-        if (need_update_table_version) {
-            std::string ver_key =
-                    table_version_key({instance_id, request->db_id(), request->table_id()});
-            txn->atomic_add(ver_key, 1);
```
Comment (deardeng) [inline cloud/src/meta-service/meta_service_partition.cpp:641]: 这里能不能新 txn_kv_->create_txn(&txn_new);，用txn_new 去自增？

File: fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java [hunk 2]
```diff
@@ -625,6 +626,10 @@ public void erasePartitionDropBackendReplicas(List<Partition> partitions) {
                 }
             }
             partitionIds.add(partition.getId());
+            if (partition.hasData()) {
```
[the remark this answers is not in the document]
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java:629]: Partition visible version has a local cache. If there is already a local cache, here we won't send a rpc.
Besides, we only need to know if the visible version is greater than 1.

File: cloud/src/meta-service/meta_service_partition.cpp (hunk 1 above)
Comment (xy720) [reply cloud/src/meta-service/meta_service_partition.cpp:641]: 这样就分成两个fdb事务了，如果前一个成功了，没有办法去rollback前一个。

Comment (deardeng) [reply cloud/src/meta-service/meta_service_partition.cpp:641]: 这个version 要求一定要精确么？

**Review by gavinchou — APPROVED** [review]

**Review by deardeng — APPROVED** [review]
LTGM

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**